tests/functional/ovncontroller_controller_test.go

```diff
@@ -689,6 +689,212 @@ var _ = Describe("OVNController controller", func() {
 		})
 	})
 
+	When("OVNController is created with networkAttachments and nic configs", func() {
+		BeforeEach(func() {
+			dbs := CreateOVNDBClusters(namespace, "")
+			DeferCleanup(DeleteOVNDBClusters, dbs)
+			name := fmt.Sprintf("ovn-controller-%s", uuid.New().String())
+			spec := GetDefaultOVNControllerSpec()
+			spec["networkAttachments"] = []string{"internalapi"}
+			spec["nicMappings"] = map[string]interface{}{
+				"physnet1": "enp2s0.100",
+			}
+			instance := CreateOVNController(namespace, name, spec)
+			DeferCleanup(th.DeleteInstance, instance)
+		})
+
+		It("reports that daemonset have annotations for both Networkattachment and nic-configs", func() {
```

Comment:
nit: have -> has, Networkattachment -> NetworkAttachment (see the A-letter case)